OpenRA.Mods.Common/Traits/Sound/VoiceAnnouncement.cs

```diff
@@ -29,21 +29,34 @@ public class VoiceAnnouncementInfo : ConditionalTraitInfo
 		[Desc("Disable the announcement after it has been triggered.")]
 		public readonly bool OneShot = false;
 
-		public override object Create(ActorInitializer init) { return new VoiceAnnouncement(this); }
+		[Desc("Wether the announcement will also be triggered by actors spawned at game start.")]
+		public readonly bool AnnounceWhenSpawnedByMap = true;
+
+		public override object Create(ActorInitializer init) { return new VoiceAnnouncement(init, this); }
 	}
 
 	public class VoiceAnnouncement : ConditionalTrait<VoiceAnnouncementInfo>
 	{
 		bool triggered;
+		bool skipNext;
 
-		public VoiceAnnouncement(VoiceAnnouncementInfo info)
-			: base(info) { }
+		public VoiceAnnouncement(ActorInitializer init, VoiceAnnouncementInfo info)
+			: base(info)
+		{
+			skipNext = !info.AnnounceWhenSpawnedByMap && init.Contains<SpawnedByMapInit>();
+		}
 
 		protected override void TraitEnabled(Actor self)
 		{
 			if (IsTraitDisabled)
 				return;
 
+			if (skipNext)
```

Comment:
Can you check against something that is actually related to MapInit instead? `self.World.WorldTick == 0` should work.